configserver/src/test/java/com/yahoo/vespa/config/server/ApplicationRepositoryTest.java

```diff
@@ -120,6 +122,24 @@ public void createAndPrepareAndActivate() {
         assertTrue(result.configChangeActions().getRestartActions().isEmpty());
     }
 
+    @Test
+    public void createFromActiveSession() {
+        PrepareResult result = deployApp(testApp);
+        long sessionId = applicationRepository.createSessionFromExisting(applicationId(),
+                                                                         new SilentDeployLogger(),
+                                                                         false,
+                                                                         timeoutBudget);
+        long originalSessionId = result.sessionId();
+        ApplicationMetaData originalApplicationMetaData = getApplicationMetaData(applicationId(), originalSessionId);
+        ApplicationMetaData applicationMetaData = getApplicationMetaData(applicationId(), sessionId);
+
+        assertNotEquals(sessionId, originalSessionId);
+        assertEquals(applicationMetaData.getApplicationName(), originalApplicationMetaData.getApplicationName());
+        assertEquals(applicationMetaData.getPreviousActiveGeneration(), originalApplicationMetaData.getGeneration().longValue());
+        assertNotEquals(applicationMetaData.getGeneration(), originalApplicationMetaData.getApplicationName());
```

Comment:
These shouldn't be equal in any case, though ;)

Reply:
Thanks, fixed